testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/ejb/timer/remote/HotRodPersistentTimerServiceTestCase.java

```diff
@@ -0,0 +1,73 @@
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.test.clustering.cluster.ejb.timer.remote;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.TargetsContainer;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.test.clustering.cluster.ejb.timer.AbstractTimerServiceTestCase;
+import org.jboss.as.test.shared.CLIServerSetupTask;
+import org.jboss.shrinkwrap.api.Archive;
+import org.junit.ClassRule;
+import org.junit.rules.TestRule;
+
+/**
+ * @author Paul Ferraro
+ */
+@ServerSetup({ InfinispanServerSetupTask.class, HotRodPersistentTimerServiceTestCase.ServerSetupTask.class })
+public class HotRodPersistentTimerServiceTestCase extends AbstractTimerServiceTestCase {
+
+    @ClassRule
+    public static final TestRule INFINISPAN_SERVER_RULE = infinispanServerTestRule();
+
+    @Deployment(name = DEPLOYMENT_1, managed = false, testable = false)
+    @TargetsContainer(NODE_1)
+    public static Archive<?> deployment0() {
+        return createArchive();
+    }
+
+    @Deployment(name = DEPLOYMENT_2, managed = false, testable = false)
+    @TargetsContainer(NODE_2)
+    public static Archive<?> deployment1() {
+        return createArchive();
+    }
+
+    private static Archive<?> createArchive() {
+        return createArchive(HotRodPersistentTimerServiceTestCase.class);
+    }
+
+    static class ServerSetupTask extends CLIServerSetupTask {
+        ServerSetupTask() {
+            this.builder.node(THREE_NODES)
+                    .setup("/subsystem=infinispan/cache-container=ejb/invalidation-cache=hotrod-persistent:add")
+                    .setup("/subsystem=infinispan/cache-container=ejb/invalidation-cache=hotrod-persistent/component=expiration:add(interval=0)")
+                    .setup("/subsystem=infinispan/cache-container=ejb/invalidation-cache=hotrod-persistent/component=locking:add(isolation=REPEATABLE_READ)")
+                    .setup("/subsystem=infinispan/cache-container=ejb/invalidation-cache=hotrod-persistent/component=transaction:add(mode=BATCH)")
+                    .setup("/subsystem=infinispan/cache-container=ejb/invalidation-cache=hotrod-persistent/store=hotrod:add(remote-cache-container=ejb, cache-configuration=default, fetch-state=false, purge=false, passivation=false, shared=true)")
+                    .setup("/subsystem=distributable-ejb/infinispan-timer-management=hotrod:add(cache-container=ejb, cache=hotrod-persistent)")
```

Comment:
I haven't run this very test yet, and couldn't check the details, but I used this very configuration in combiantion with the one defined by the `InfinispanServerSetupTask` and noticed that the timers were persisted locally on the file system, i.e. according to the default server behavior.

To make it persisted on a remote Infinispan, I had to follow what in the community docs to enable the feature explicitly, see https://github.com/wildfly/wildfly/blob/b9d1aa85531fa0afa1dd59aed2be72e4d7a8782d/docs/src/main/asciidoc/_developer-guide/ejb3/Jakarta_Enterprise_Beans_Distributed_Timers.adoc#setup which seems correct to me, as it is mentioned in the proposal as well.

So this is JFYI in case you think a green bar could be hiding this behavior for you as well, otherwise if the missing piece of configuration is actually defined somewhere else in here and executed, please could you point me at such place?

Reply:
This setup script does not change the default persistent timer management of the server.